File: java/src/main/java/dev/blackoutburst/shitvm/Frame.java
package dev.blackoutburst.shitvm;

import java.util.ArrayList;
import java.util.List;

public class Frame {
    VClass vclass;
    int ip;
    byte[] code;
    List<Object> locals;
    List<Object> stack;

    public Frame(VClass vclass, int ip, byte[] code) {
        this.vclass = vclass;
        this.ip = ip;
        this.code = code;
        this.locals = new ArrayList<>();
        this.stack = new ArrayList<>();
    }
}

File: js/shitjsvm/js/class/Frame.js
export class Frame {
    constructor(maxStack, maxLocal, codeLength, jClass, code) {
        this.maxStack = maxStack
        this.maxLocal = maxLocal
        this.jClass = jClass
        this.codeLength = codeLength
        this.ip = 0
        this.code = code
        this.locals = []
        this.stack = []
    }
}
